### PR TITLE
Improve type inference

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
The TimeHelper test failure is unrelated to these changes. Seems it's due to some edge case for relative time calculation.

File: src/Mailer/TransportFactory.php [hunk 1]
```diff
@@ -71,10 +71,10 @@ public static function setRegistry(TransportRegistry $registry): void
      * Finds and builds the instance of the required transport class.
      *
      * @param string $name Name of the config array that needs a transport instance built
-     * @return void
+     * @return \Cake\Mailer\AbstractTransport
      * @throws \InvalidArgumentException When a transport cannot be created.
      */
-    protected static function _buildTransport(string $name): void
+    protected static function _buildTransport(string $name): AbstractTransport
```
Comment (LordSimal) [inline src/Mailer/TransportFactory.php:77]: Is it expected, that users overwrite this method?

Comment (ADmad) [reply src/Mailer/TransportFactory.php:77]: There's no BC guarantee for protected methods in a new minor. Also the probability of someone extending the TransportFactory are negligible.

**Review by LordSimal — APPROVED** [review]
